### PR TITLE
Logging / get_geo_list diverse fixes

### PR DESCRIPTION
**Pull request by tgrandje**
I haved replaced the previous PR. This should allow :

- to set the logging level beforehand (there was a glitch before) ;
- fix issues on get_geo_list (https://github.com/InseeFrLab/pynsee/issues/140 https://github.com/InseeFrLab/pynsee/issues/139 https://github.com/InseeFrLab/pynsee/issues/138 and https://github.com/InseeFrLab/pynsee/issues/136)

### CODE REVIEW

**Comment by tgrandje** [comment]
Note that the logging is actually totally standard. You can active a "verbose" mode by using the following syntax :
```
import logging
logging.basicConfig(level=logging.DEBUG)
from pynsee.localdata import get_geo_list
df = get_geo_list("regions", "2023-01-01", update=True)
```

File: pynsee/localdata/_get_geo_list_simple.py [hunk 1]
```diff
@@ -12,9 +12,10 @@
 
 
 @lru_cache(maxsize=None)
-def _get_geo_list_simple(geo, progress_bar=False):
-
+def _get_geo_list_simple(geo, date=None, progress_bar=False):
```
Comment (tfardet) [inline pynsee/localdata/_get_geo_list_simple.py:15]: shouldn't this be in a separate PR? this one is already quite big (for intrinsic reasons) and this change is drowned in unrelated changes.

File: pynsee/localdata/get_geo_list.py [hunk 2]
```diff
@@ -19,6 +24,8 @@ def get_geo_list(geo=None, date=None, update=False):
     Args:
         geo (str): choose among : communes, communesDeleguees, communesAssociees, regions, departements, arrondissements, arrondissementsMunicipaux
 
+        date (str): date of validity (AAAA-MM-DD)
```
Comment (tfardet) [inline pynsee/localdata/get_geo_list.py:27]: same remark

**Review by tfardet — CHANGES_REQUESTED** [review]
Thanks for opening a new PR for logging since the PR seemed stalled.
I would recommend to split the ``get_geo_list`` part out into a new PR. Mixing both makes it impractical to see and discuss the latter.
